src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -525,23 +745,44 @@ public void AssemblyLoading_MultipleVersions_MultipleEqualMatches()
                 var e = epsilon.CreateInstance("Epsilon.E")!;
                 e.GetType().GetMethod("Write")!.Invoke(e, new object[] { sb, "Test E" });
 
-                VerifyDependencyAssemblies(
-                    loader,
-                    testFixture.Delta2.Path,
-                    testFixture.Epsilon.Path);
+                if (ExecutionConditionUtil.IsDesktop && loader is ShadowCopyAnalyzerAssemblyLoader)
+                {
+                    // Delta2B and Delta2 have the same version, but we prefer Delta2B because it's added first and 
+                    // in shadow loader we can't fall back to same directory because the runtime doesn't provide
```

Comment:
Is this something that would be alleviated if the shadow loader grouped its outputs based on the directory structure of the "original" assemblies?